### DISCUSSION TITLE
When will plugins allow UI alterations?

### DISCUSSION THREAD

**Discussion by J-Cake** [lapce/lapce#2515] [General]
One of the things I find most frustrating about Lapce is the lack of a task-runner system like in VSCode, but in my opinion, not many editors (including VSCode) manage to get this tasks system right, so I was hoping to create my own for Lapce, but it seems that plugins don't support this. I was hoping someone could advise me on how/when plugins offer the ability to create custom panels/dialogs/editors etc. to allow me to begin implementing a tasks system?

Thanks

**Comment by panekj** [comment]
> I was hoping someone could advise me on how/when plugins offer the ability to create custom panels/dialogs/editors etc. to allow me to begin implementing a tasks system?

It's not going to happen anytime soon

**Reply by J-Cake** [reply]
Ah... well that's disheartning...

**Reply by J-Cake** [reply]
Do we also know why?

**Reply by panekj** [reply]
There is no one working full time on Lapce, it's not possible to say when it will happen

**Reply by panekj** [reply]
But task running is already implemented in Lapce and will be available in next release

**Reply by J-Cake** [reply]
Ah I see. Theoretically then, if I were to make the necessary changes to Lapce, is there a chance that they'll get merged?

**Reply by panekj** [reply]
It's better to state what exactly you want to achieve

**Reply by J-Cake** [reply]
I'd like to implement a graphical way to define tasks and their dependencies. These tasks will be in the form of shell code, with the option to specify things like output parsers and concurrent tasks. I'd need to be able to create things like rich text editors, dropdowns, notifications, and edit the status bar.
